translator/navjoy_translator.py

```diff
@@ -67,21 +121,49 @@ def wzdx_creator(messages, info=None, unsupported_message_callback=None):
     wzd = wzdx_translator.initialize_wzdx_object(info)
 
     for message in messages:
-        # closures can cover both directions
-        directions = message.get('data', {}).get('directionOfTraffic')
 
-        for direction in get_directions_from_string(directions):
-            # Parse closure to WZDx Feature
+        separated_messages = expand_speed_zone(message)
+        for msg in separated_messages:
             feature = parse_reduction_zone(
-                copy.deepcopy(message), direction, callback_function=unsupported_message_callback)
+                copy.deepcopy(msg), callback_function=unsupported_message_callback)
```

Comment:
Do we need to `deepcopy` the msg here? We're already working with a copy and don't seem to use the value after this parse method anyway.

Reply:
I was having some initial pass by reference issues, so I added in deepcopies everywhere. I will take a look and see which ones are actaully necessary

Reply:
Just removed this